lib/process-services-cloud/src/lib/form/components/widgets/dropdown/dropdown-cloud.widget.ts

```diff
@@ -74,23 +76,89 @@ export class DropdownCloudWidgetComponent extends WidgetComponent implements OnI
     }
 
     ngOnInit() {
-        if (this.field.restUrl && !this.isLinkedWidget()) {
-            this.persistFieldOptionsFromRestApi();
+        this.checkFieldOptionsSource();
+        this.updateOptions();
+    }
+
+    private checkFieldOptionsSource(): void {
+        switch (true) {
+            case this.field.restUrl && !this.isLinkedWidget():
+                this.persistFieldOptionsFromRestApi();
+                break;
+
+            case this.isLinkedWidget():
+                this.loadFieldOptionsForLinkedWidget();
+                break;
+
+            case this.isVariableOptionType():
+                this.persistFieldOptionsFromVariable();
+                break;
+
+            default:
+                break;
         }
+    }
 
-        if (this.isLinkedWidget()) {
-            this.loadFieldOptionsForLinkedWidget();
-
-            this.formService.formFieldValueChanged
-                .pipe(
-                    filter((event: FormFieldEvent) => this.isFormFieldEventOfTypeDropdown(event) && this.isParentFormFieldEvent(event)),
-                    takeUntil(this.onDestroy$))
-                .subscribe((event: FormFieldEvent) => {
-                    const valueOfParentWidget = event.field.value;
-                    this.parentValueChanged(valueOfParentWidget);
-                });
+    private persistFieldOptionsFromVariable(): void {
+        const optionsId = this.field?.variableConfig?.optionsId ?? 'id';
+        const optionsLabel = this.field?.variableConfig?.optionsLabel ?? 'name';
+        const optionsPath = this.field?.variableConfig?.optionsPath ?? 'data';
+        const variableId = this.field?.variableConfig?.variableId;
+
+        const formVariables: TaskVariableCloud[] = this.field?.form?.variables;
+        const dropdownOptions: TaskVariableCloud = formVariables.find((variable: TaskVariableCloud) => variable?.id === variableId)?.value;
+
+        if (dropdownOptions) {
+            const formVariableOptions: FormFieldOption[] = this.getOptionsFromPath(dropdownOptions, optionsPath, optionsId, optionsLabel);
+            this.field.options = formVariableOptions;
+            this.resetInvalidValue();
+            this.field.updateForm();
+        } else {
+            this.handleError(`${variableId} not found in ${JSON.stringify(formVariables)}`);
+            this.resetOptions();
+            this.variableOptionsFailed = true;
         }
-        this.updateOptions();
+    }
+
+    private getOptionsFromPath(data: any, path: string, id: string, label: string): FormFieldOption[] {
+        const properties = path.split('.');
```

Comment:
maybe this method is a bit too long and it can be split

Reply:
splitted to `getOptionsFromPath`, `getOptionsFromArray` and `createOption`